check_whitespace.c

```diff
@@ -66,8 +67,14 @@ int is_clean(char* str) {
   // 0 if they're equal, and a positive value if the first is
   // greater than the second.
   result = strcmp(str, cleaned);
+  
+  // Makes sure that cleaned() has memory to free
+  if(cleaned != ""){
```

Comment:
If you compile this with the `-Wall` flag you should get something like:

```
check_whitespace.c:72:14: warning: result of comparison against a string literal is unspecified (use strncmp instead) [-Wstring-compare]
  if(cleaned != ""){
             ^  ~~
```

This is telling you that you don't want to use `==` or `!=` on strings, but should instead use the `strcmp` function, as I've done below:

```suggestion
  if(strcmp(cleaned, "") != 0) {
```